variables.tf

```diff
@@ -12,6 +12,10 @@ variable "server_version" {
   default     = "12.0"
 }
 
+variable "server_name" {
+  description = "The name of the server to be created."
```

Comment:
Are there any specific reasons not putting default value here?

Reply:
I think users should choose a server name by themselves because it should be globally unique. Thus, if users choose default value, it will easily cause a name conflict. But it's also totally ok to have a default value for server name, just as resource group name.

Reply:
My concern is, if you do not provide a default name, actually all legacy code would not work any more.

Reply:
Please update the PR with default value.